plugins/inlinetoolbar/plugin.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or http://ckeditor.com/license
+ */
+
+( function() {
+	'use strict';
+
+	/**
+	 * A class that represents a view of inline toolbar capable of presenting defined
+	 * menu buttons.
+	 *
+	 * @class
+	 * @extends CKEDITOR.ui.balloonPanel
+	 * @constructor Creates a inline toolbar view instance.
+	 * @since 4.8
+	 * @param {CKEDITOR.editor} editor The editor instance for which the panel is created.
+	 * @param {Object} definition An object containing the panel definition.
+	 */
+	CKEDITOR.ui.inlineToolbarView = function( editor, definition ) {
+		CKEDITOR.ui.balloonPanel.call( this, editor, definition );
+		this.listeners = [];
+	};
+
+		/**
+	 * A class that represents a inline toolbar capable of presenting defined
+	 * menu buttons.
+	 *
+	 * @class
+	 * @since 4.8
+	 * @param {CKEDITOR.editor} editor The editor instance for which the panel is created.
+	 */
+	CKEDITOR.inlineToolbar = function( editor ) {
+		this.inlineToolbar = new CKEDITOR.ui.inlineToolbarView( editor );
+	};
+
+	CKEDITOR.plugins.add( 'inlinetoolbar', {
+		requires: 'balloonpanel',
+		init: function() {
+			CKEDITOR.ui.inlineToolbarView.prototype = CKEDITOR.tools.extend( {}, CKEDITOR.ui.balloonPanel.prototype );
+			/**
+			 * Build inline toolbar DOM representation.
+			 */
+			CKEDITOR.ui.inlineToolbarView.prototype.build = function() {
```

Comment:
Here no but next PR is using this to rebuild button structure. So after all we will overwrite this.

Reply:
Ok, I see. It might be hard to code the same functionality in 3 separated PRs :) While this code is correct, and my proposal was only a simplification it might stay as it is.